### PR TITLE
Added Throwable Stacktrace to ShouldHaveCauseExactlyInstance

### PR DESCRIPTION
**Pull request by shaikhu**
#### Check List:
* Fixes #1355 
* Unit tests : YES (no new)
* Javadoc with a code example (on API only) : NA
* PR meets the [contributing guidelines](https://github.com/assertj/assertj/blob/main/CONTRIBUTING.md)

Following the [contributing guidelines](https://github.com/assertj/assertj/blob/main/CONTRIBUTING.md) will make it easier for us to review and accept your PR.

I'd like to help get this over the line 😄. I used this [commit](https://github.com/assertj/assertj/commit/9739d6995940b3cef37c9c99e6bd3d62ea84b704) as a guide/reference. Seems most of the changes where already done.

### CODE REVIEW

**Comment by shaikhu** [comment]
@scordio @joel-costigliola All tests are passing now. Any chance either of you can have a look?

File: assertj-core/src/main/java/org/assertj/core/error/ShouldHaveCauseExactlyInstance.java [hunk 1]
```diff
@@ -47,8 +47,8 @@ private ShouldHaveCauseExactlyInstance(Throwable actual, Class<? extends Throwab
           expectedCauseType, actual.getCause().getClass());
   }
 
-  private ShouldHaveCauseExactlyInstance(Class<? extends Throwable> expectedCauseType) {
-    super("%nExpecting a throwable with cause being exactly an instance of:%n  %s%nbut current throwable has no cause.",
-          expectedCauseType);
+  private ShouldHaveCauseExactlyInstance(Class<? extends Throwable> expectedCauseType, Throwable actual) {
+    super("%nExpecting a throwable with cause being exactly an instance of:%n  %s%nbut current throwable has no cause." +
+          "%nThrowable that failed the check:%n" + escapePercent(getStackTrace(actual)), expectedCauseType);
```
[the remark this answers is not in the document]
Comment (shaikhu) [reply assertj-core/src/main/java/org/assertj/core/error/ShouldHaveCauseExactlyInstance.java:52]: > perf nit: should this defer the overhead of stacktrace string generation and concatenation for only the failure paths?
> 
> For example, something like:

TBH I'm not sure any performance gain is worth the sacrifice in readability? I'd also like to keep consistent with [ShouldHaveRootCauseExactlyInstance](https://github.com/assertj/assertj/blob/07d1c9828e0b6b6f2d276ae5585c2812a9317854/assertj-core/src/main/java/org/assertj/core/error/ShouldHaveRootCauseExactlyInstance.java#L50), but can make the change if you feel prefer.

[the remark this answers is not in the document]
Comment (joel-costigliola) [reply assertj-core/src/main/java/org/assertj/core/error/ShouldHaveCauseExactlyInstance.java:52]: I agree with @shaikhu here, I don't see the performance impact to be significant enought.

[the remark this answers is not in the document]
Comment (shaikhu) [reply assertj-core/src/main/java/org/assertj/core/error/ShouldHaveCauseExactlyInstance.java:52]: @joel-costigliola @schlosna Do either of you have any other concerns?

**Comment by joel-costigliola** [comment]
Integrated thanks @shaikhu!

**Pull request closed without merging**